lab-judy/server.js

```diff
@@ -0,0 +1,81 @@
+'use strict';
+
+const net = require('net');
+const EE = require('events');
+
+
+
+const Client = require('./model/client.js');
+
+const PORT = process.env.PORT || 3000;
+
+const pool = [];
+const nickNamesArray = [];
+const server = net.createServer();
+const ee = new EE();
+
+ee.on('\\user', function(client, string){
+  client.nickname = string.trim();
+  nickNamesArray.push(client.nickname);
+  console.log(nickNamesArray);
+});
+
+
+ee.on('\\all', function(client, string){
+  pool.forEach( c => {
+    c.socket.write(`${client.nickname}: ` + string);
+  });
+});
+
+ee.on('\\showusers', function(){
+  console.log(pool, ' line 28');
```

Comment:
Keep in mind that console.log will only show up on the server, so this event has no use to the clients. But it doesn't hurt anything. For it to show up for the clients, you would need to ***write*** to each socket in the pool.